api/clients/disperser_client.go

```diff
@@ -26,6 +26,14 @@ type Config struct {
 	// TODO: do we want to add config timeouts for those separate requests?
 	Timeout           time.Duration
 	UseSecureGrpcFlag bool
+	// MaxRetrieveBlobSizeBytes is the maximum size of a blob that can be retrieved by using
+	// the RetrieveBlob method. This is used to set the max message size for the grpc client.
+	// DisperserClient uses a single underlying grpc channel shared for all methods,
+	// but all other methods use the default 4MiB max message size, whereas RetrieveBlob
```

Comment:
I think dispersal and retrieval for a blob have the same blob length.

It is just the receiving ends are different. On our dispersal endpoint, we also set a very large buffer size. Then similarly, if you retrieve, you should set a large buffer. The buffer size on the sender side does not matter

Reply:
It does, both sides set their respective length that they want.
<img width="701" alt="image" src="https://github.com/user-attachments/assets/f36ed91b-1a39-46d2-be8c-50a478c73a12">

Disperser needs large buffer to receive large blobs for dispersal, and client needs large buffer size to retrieve blobs.